Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml

```diff
@@ -16915,4 +16926,31 @@
         <EffectArray Operation="Set" Reference="Button,AP_BanelingLaunch,Tooltip" Value="Button/Tooltip/AP_BanelingLaunchLevel2"/>
         <EffectArray Operation="Set" Reference="Abil,AP_BanelingLaunchHangar,InfoArray[Ammo1].Button.Requirements" Value="AP_BanelingLaunchArmLevel2"/>
     </CUpgrade>
+    <CUpgrade id="AP_ArchonFloat">
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Mover" Value="Colossus"/>
+        <!-- 1 = ground, 8 = forcefield, 15 = small, 4 = colossus, 5 = structure -->
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Collide[1]" Value="0"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Collide[8]" Value="0"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Collide[15]" Value="0"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Collide[4]" Value="1"/>

```

Comment:
Unupgraded don't collide with colossi, the upgrade could bring you into problems with using such units (a possible downgrade)

Reply:
Changed so floating Archons get collision flag 12, which is only shared with Reaper charges. Best ground collision flag I could find, so they still block each other.